### ISSUE TITLE
Backup & Restore not working, Tried Centos, Debian, Ubuntu => nothing [Deployment Issue]

### ISSUE THREAD

**Issue by ph0** [caprover/caprover#1636]
I followed this to a T on Centos, on Debian, and on Ubuntu - https://caprover.com/docs/backup-and-restore.html
and when I go to http://[myIp]:3000 I get
`Secure Connection Failed

An error occurred during a connection to [myip]:3000. SSL received a record that exceeded the maximum permissible length.

Error code: `SSL_ERROR_RX_RECORD_TOO_LONG``

What is notable here is that simply doing a fresh blank install of caprover worked perfectly fine.

I am at a loss. I think the feature is broken.

The troubleshooting documentation even goes to a "page not found" link.

**Comment by githubsaturn** [comment]
When you restore your instance, you don't need to use the port 3000 anymore. If you were using captain.domain.com before, after restoration, you can use the same domain.

If you're still having issues, please provide a step by step steps on how to reproduce the issue. I just retried taking a backup and restore, and it just worked fine.


> The troubleshooting documentation even goes to a "page not found" link.

Can you specify what link you're referring to?

**Comment by ph0** [comment]
Thank you sooo much for your response. I realize I am probably doing it wrong so I will do my best to state my problem.

1. I have Server A with my old Caprover Content hosting my websites @ [XYZ IP address]
2. I need to move everything to Server B and close server A
3. My DNS for Server A, [A Record] is       *.server    

Questions
Do I make my [A Record] for server B           *.server    ?
My backup file shows *.server for my caprover items.
Do I need to make Server B, A record, *.move  
so when I complete transfer I can change it back to *.server so there are no conflicts while both servers are live?

I did *.s4 for example on Server b and because [myip] and [myip:3000]was not working then nothing at all even displayed at my *.s4

-----
[Connection Timeouts](https://caprover.com/docs/troubleshooting.html) links to this https://training.mirantis.com/article/ipvs-connection-timeout-issue

To recap.
Server A with caprover needs to be closed down and Server B is the new home for caprover.

I install new caprover and caprover works perfectly.

I reinstall my Ubuntu and try again with Backup & Restore method then my Server B IP does not show caprover on any link whatsoever, should it not display at 3000 to begin with before I DNS? It detects my backup and says sucessful but there is no caprover for me to continue the process

**Comment by ph0** [comment]
Right now I am attempting again by first deleting my old *.server from Server A so the only A record now is for Server B with *.server

**Comment by ph0** [comment]
I am getting 1001 : Captain is not ready yet...
when I try to login now. 

Okay deleting the old A record first was the key here to giving me a captain to see. I think this shuld be included int he documentation, it seems to be implied here with "You need to configure your DNS such that *.youroldroot.domain.com points to the new server IP."
But I just spent 20 hours the last 4 days trying to realize that.

"Configure such that" ? Delete A record from server 1 so A record from Server b serves you the new Caprover. Something like that would be a lot better.

**Comment by ph0** [comment]
After about 10 minutes I have successfully logged in and the backup and restore looks complete. Please keep a record of this train of thought because from googling around I know others have this issue because the documentation is a stumbling block with the "configure such that" part that is not express on deleting A record from server 1.

**Comment by ph0** [comment]
Lastly, how do i "redeploy all my apps" ? I could not find this documentation.

**Comment by githubsaturn** [comment]
Backup and restore assumes you are restoring on the same server, but there is a section in the doc dedicated to [Keeping Old Server
](https://caprover.com/docs/backup-and-restore.html#keeping-old-server)

If you follow the guide there, you don't need to remove the old A record from DNS, as your local /etc/hosts will override that.



>  Lastly, how do i "redeploy all my apps" ? I could not find this documentation.

![image](https://user-images.githubusercontent.com/10582082/212490401-405fd64c-f003-4283-b388-7f0569fdb4a3.png)


If you want to manually redeploy all your apps:
- If you build an app manually using `caprover deploy` - they you can simply run `caprover deploy` again and deploy it.
- If you used a one click app that uses a public registry image, e.g. `mysql:5`, you don't need to do anything, this image is available in the new server.
- If you use a one click app, but the image is a customized image (img-captain--something:456), then you can use [docker save](https://docs.docker.com/engine/reference/commandline/save/) and `docker load` to export the image from the old server and import it on the new server.

**Comment by ph0** [comment]
Bare with me I am not always the sharpest crayon in the box but I have a big heart. Please excuse my boring and processional questions as genuine inquiry for understanding.

Am I understanding correctly that because I went to a new server the new server already redeployed my one-click apps including my databases and if not then how do I perform the task for each one?

What does it mean for my image to be a customized image if it's a one click-app? Is it that the image name has a custom name or that you are illustrating a custom port with the 456?

Thank you for your help you are my shining light in my day of darkness.

**Comment by githubsaturn** [comment]
> Am I understanding correctly that because I went to a new server the new server already redeployed my one-click apps including my databases and if not then how do I perform the task for each one?

As I said it depends on the type of deployment. Go to the deployment tab. If the image name starts with `img-captain--`, you need to manually re-deploy your app. If the image name is something like any of the following, you don't need to do anything:
- `mysql`
- `portrainer/webui`
- `index.docker.io/facebook/textai`

These images are hosted on remote registries (docker hub) and Docker automatically pulls them to your new server.

But if the image starts with `img-captain` it means that the image was built on the old server and cannot be automatically redeployed to the new server.


Also, keep in mind that CapRover backup does NOT back up your data (for example, your database data is not backed up). You need to do this separately. It's called out in the doc:
https://caprover.com/docs/backup-and-restore.html#what-is-restored

In general, these are advanced concepts. If you are not too familiar with these, you can either build a new server and ignore the restoration process, or spent some time reading on how docker volumes and images work (it's a complex subject and takes weeks, if not months, to master), example book: https://www.amazon.com/Docker-Shipping-Reliable-Containers-Production

**Comment by ph0** [comment]
Perhaps I was wrong about having success. I switched my DNS to my new server ip and I get a secure connection failed "Error code: SEC_ERROR_EXPIRED_CERTIFICATE'

**Comment by ph0** [comment]
I think I have to build a new caprover and basically copy paste all of my websites. I am on a dedicated server of my own.. Unfortunately while my host was very helpful as a new customer and resolved way more than this issue at the start .. I am treated the opposite now.

**Comment by ph0** [comment]
Thank you for your help, I learned a lot about some and more of what I must learn!

**Comment by githubsaturn** [comment]
I am sorry that I cannot help further. It's not practical for a free/open source software to dedicate the same support level as a paid service. If you'd like a dedicated support channel, there are a lot of folks who are available to offer you their services on our Slack channel, or here: https://caprover.com/docs/support.html#need-hourly-consultation

**Comment by ph0** [comment]
I am sorry, I am finished with my silly questions. I have a clear path which is a perfect resolution for me. I am on a fresh Caprover working fine and deploying my 1 clicks, exporting my content from old server, and reupping, all is perfect in the world - I had not realized there would be knowledge discrepancies between me and database and images and the like. I was thinking maybe I could just take my backup and add it to a fresh cap install, maybe in the future. Have a good one and thanks again you really did give me the clarity I was after.
